### ISSUE TITLE
Outdated SoS conda packages

### ISSUE THREAD

**Issue by gaow** [vatlab/sos#1544]
Here is the list of outdated packages on conda-forge that we would need to fix, comparing pypi vs conda-forge:

- https://anaconda.org/conda-forge/jupyterlab-sos Not all platforms have the same version of build. Need a fix. 

- https://github.com/conda-forge/jupyterlab-transient-display-data-feedstock is 0.4.3, is 0.4.4 at https://pypi.org/project/transient-display-data/, should be 0.5.3 based on GitHub commit message 
- https://github.com/conda-forge/sos-bash-feedstock is 0.20.0, should be 0.21.0 at https://pypi.org/project/sos-bash/
- https://github.com/conda-forge/markdown-kernel-feedstock is 0.2.1, should be 0.2.2 https://pypi.org/project/markdown-kernel/
- https://github.com/conda-forge/sos-matlab-feedstock is 0.18.8, should be 0.20.3
- https://github.com/conda-forge/sos-rmarkdown-feedstock is 0.1.3, should be 0.1.4 
- https://github.com/conda-forge/sos-python-feedstock is 0.18.4, should be 0.18.5
- https://github.com/conda-forge/sos-pbs-feedstock is 0.20.8, should be 0.20.9

**Comment by gaow** [comment]
@BoPeng we need Jupyterlab-sos to be the most up-to-date on conda for our production workflow. This PR actually seem to work, https://github.com/conda-forge/jupyterlab-sos-feedstock/pull/24 could you review and merge? My other PR where I bumped version requirements did not work due to failed tests.

**Comment by gaow** [comment]
Just to let you know that we are taking a look into this with help from @danielnachun, hopefully pushing a fix soon.

**Comment by danielnachun** [comment]
An important update here - many of the feedstocks are out of date because the issue with `ipkernel.tests` identified in https://github.com/vatlab/sos-notebook/issues/354.  As of now I believe the follow `conda-forge` feedstocks are affected:

- `sos-bash`
- `sos-julia`
- `sos-matlab`
- `sos-notebook`
- `sos-python`
- `sos-r`

Of the remaining outdated feedstocks:
- `sos-rmarkdown` and `sos-pbs` have no apparent issues and I'll rerender those
- `markdown-kernel` has a linting issue with its recipe which I will fix in the update PR.  
- `jupyterlab-transient-display-data` had some missing dependencies and an issue with its source URL, which I will push a fix for.
- `jupyterlab-sos` had some missing dependencies and an issue with properly installing its extension, as discussed here: https://github.com/vatlab/jupyterlab-sos/issues/72.  See that issue for more details on the fixes.

**Comment by BoPeng** [comment]
For the kernels, maybe we can comment out the tests and check only if the modules can be imported.

**Comment by gaow** [comment]
Okay it sounds like you would remove those tests and make new releases? Still there might be some issues that @danielnachun fixed. Perhaps we should PR those other fixes first, and assuming once the tests issues are addressed and new releases is out, the conda package will be automatically updated? I think @danielnachun also mentioned that we should allow for auto-merge after all tests pass, to save us some effort. I take that those will be implemented too in the PRs.

**Comment by danielnachun** [comment]
@BoPeng I think all the kernel recipes already have an import check.  For most of them we'll just end up deleting the `run_test.py` file from the feedstock as its only purpose is to run that unit test.  I believe the only exception to this is `sos-notebook` which has some other test in that script, so I'll just delete the part that needs `ipykernel.test`.

**Comment by danielnachun** [comment]
@BoPeng I have made the PRs to address all but one feedstock.  Can you merge the following PRs:
- https://github.com/conda-forge/jupyterlab-sos-feedstock/pull/26
- https://github.com/conda-forge/jupyterlab-transient-display-data-feedstock/pull/8
- https://github.com/conda-forge/sos-bash-feedstock/pull/11
- https://github.com/conda-forge/sos-notebook-feedstock/pull/87
- https://github.com/conda-forge/sos-python-feedstock/pull/10
- https://github.com/conda-forge/sos-rmarkdown-feedstock/pull/4
- https://github.com/conda-forge/sos-matlab-feedstock/pull/15
- https://github.com/conda-forge/markdown-kernel-feedstock/pull/3
- https://github.com/conda-forge/sos-r-feedstock/pull/9
- https://github.com/conda-forge/sos-julia-feedstock/pull/9

I will create a separate set of PRs after these are merged to add myself as a feedstock maintainer, because right now the CI would fail for most of the feedstocks.  

The only remaining issue now is I don't see a version 0.20.9 of `sos-pbs` on PyPI - the newest version is 0.20.8, the same as in conda-forge.  If a version 0.20.9 does exist, it will need to be uploaded to PyPI before we can update in conda-forge.  Otherwise if 0.20.8 is newest version we'll just want to get auto-merge enabled, which I will make a new PR for.

**Comment by BoPeng** [comment]
Thanks. I have merged the PRs and released sos-pbs 0.20.9. It contains a few bug fixes that I am not sure if it is the fault of sos-pbs or my local configuration, but I made the release anyway.

**Comment by danielnachun** [comment]
Thank you so much!  I made a PR for `sos-pbs` 0.20.9 - can you merge it as well: https://github.com/conda-forge/sos-pbs-feedstock/pull/14. 

The only remaining question brought up by @gaow was the version of `jupyterlab-transient-display-data`.  The latest version on PyPI is 0.4.4, but @gaow mentioned seeing version 0.5.3 in commit messages.  Is there a plan to release a new version on PyPI.  Now that we have auto-merge enabled for the feedstock, it should update itself automatically after a new PyPI release.

**Comment by gaow** [comment]
Thanks! To clarify, we should not have arch specific release for these packages right? Still here for example: https://anaconda.org/conda-forge/sos-notebook I see the noarch is up to date but there are arch specific releases.

**Comment by BoPeng** [comment]
Regarding transient-display-data, a commit message for the release of version 0.4.3 has a tupo for version 0.5.3. Then version 0.4.4 for Jlab 4.x was released without an explicit "release" message. The current pypi (and conda) release of 
0.4.4 should work.

**Comment by danielnachun** [comment]
> Thanks! To clarify, we should not have arch specific release for these packages right? Still here for example: https://anaconda.org/conda-forge/sos-notebook I see the noarch is up to date but there are arch specific releases.

Both conda-forge and bioconda don't allow the deletion of old packages unless they contain malicious code or constitute license violations.  So the arch-specific packages won't disappear, but won't be updated anymore.  They can just be ignored  - the only time they'd be an issue is if the user wanted to use the old version that was still arch specific on a platform that didn't support it, and I don't think for this type of software someone is going to need to do that.

**Comment by danielnachun** [comment]
> Regarding transient-display-data, a commit message for the release of version 0.4.3 has a tupo for version 0.5.3. Then version 0.4.4 for Jlab 4.x was released without an explicit "release" message. The current pypi (and conda) release of 0.4.4 should work.

Thanks @BoPeng for clarifying this, in that case we are good to go for `transient-display-data`.  All of the outdated `conda-forge` packages are now fully up to date with what is on PyPI, so I think we can close this issue.  I am going to open a separate set of PRs tomorrow to add myself as a feedstock maintainer for these feedstocks, but that won't have any affect on the packages.  Once those PRs are merged, I'll be able to merge another set to add @gaow as a maintainer and then the 3 of us will all be to help maintain the feedstocks for the foreseeable future.

**Comment by BoPeng** [comment]
Thanks for taking care of this mass.
